### PR TITLE
add byteman package

### PR DESCRIPTION
**Pull request by kranurag7**
required for hadoop package. 

#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [x] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [x] REQUIRED - The version of the package is still receiving security updates
- [ ] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

### CODE REVIEW

File: byteman.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+# NOTE: JRE runtime dependency required but only jar is needed at this point for using this package as part of hadoop.
+environment:
+  contents:
+    packages:
+      - busybox
+      - ca-certificates-bundle
+      - zip
+      - unzip
+      - maven
+      - openjdk-9
```
Comment (EyeCantCU) [inline byteman.yaml:18]: Is `openjdk-9` a typo?

Comment (kranurag7) [reply byteman.yaml:18]: RJ, from upstream reference https://github.com/bytemanproject/byteman/blob/4e8bd4985ed3110ff0a2e226e154a72c0d7b8579/bin/bminstall.sh#L40-L52 I found out that only two versions are supported for building it. I first tried it with openjdk-8 and that fails. Later I switched to using openjdk-9 here. 

I also checked the docs but was not able to find supported openjdk version. (ref: https://downloads.jboss.org/byteman/4.0.23/byteman-programmers-guide.html#building-byteman-from-the-sources)

File: byteman.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+test:
+  environment:
+    contents:
+      packages:
+        - openjdk-9
+        - openjdk-9-default-jvm
+  pipeline:
+    - name: jar listing
+      runs: |
+        jar -tf /usr/share/java/byteman/byteman.jar
```
Comment (jamonation) [inline byteman.yaml:63]: Is there a way to run the main class in the jar, or import it in a test and run a program that relies on it?

Comment (kranurag7) [reply byteman.yaml:63]: Thanks Jamon, I've addressed this via https://github.com/wolfi-dev/os/pull/29349/commits/0bb32bc9479af2774d5aa9f552b3629dba62c5f1 (it runs a simple hello world app using byteman agent)

Comment (jamonation) [reply byteman.yaml:63]: Terrific, that's exactly the kind of test I was hoping for. Now we know it works as intended right at package build time.

**Review by jamonation — DISMISSED** [review]

File: byteman.yaml (hunk 1 above)
Comment (EyeCantCU) [reply byteman.yaml:18]: JDK 9 is EOL, let's use 11 as it supports 9+

File: byteman.yaml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+environment:
+  contents:
+    packages:
+      - busybox
+      - ca-certificates-bundle
+      - zip
+      - unzip
+      - maven
+      - openjdk-9
+      - openjdk-9-default-jvm
```
Comment (EyeCantCU) [inline byteman.yaml:19]: We can remove this

File: byteman.yaml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  github:
+    identifier: bytemanproject/byteman
+    use-tag: true
+
+test:
+  environment:
+    contents:
+      packages:
+        - openjdk-9
+        - openjdk-9-default-jvm
```
Comment (EyeCantCU) [inline byteman.yaml:59]: And this

**Review by EyeCantCU — CHANGES_REQUESTED** [review]

File: byteman.yaml (hunk 4 above)
Comment (EyeCantCU) [reply byteman.yaml:59]: Confirmed the JVM is actually used in tests

**Review by EyeCantCU — APPROVED** [review]

**Pull request merged**